### ISSUE TITLE
   @DynamoDBIndexHashKey(globalSecondaryIndexName nullpointexception

### ISSUE THREAD

**Issue by rafaelhiguerasvargas** [derjust/spring-data-dynamodb#81]
Hello, when i try a query by GSI, i have a nullpointerexception from amazon API 

```
java.lang.NullPointerException
	at com.amazonaws.services.dynamodbv2.datamodeling.StandardModelFactories.rulesOf(StandardModelFactories.java:157)
	at com.amazonaws.services.dynamodbv2.datamodeling.StandardModelFactories.access$100(StandardModelFactories.java:53)
	at com.amazonaws.services.dynamodbv2.datamodeling.StandardModelFactories$StandardModelFactory.getTableFactory(StandardModelFactories.java:80)
	at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.getTableModel(DynamoDBMapper.java:393)
	at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.toParameters(DynamoDBMapper.java:1960)
	at com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.toParameters(DynamoDBMapper.java:1949)
```

My code;
```java
public interface FacturaRepository extends DynamoDBPagingAndSortingRepository<Factura, FacturaId> {
List<Factura> findByClaveFactura(String claveFactura);
}
```
```java
 @DynamoDBIndexHashKey(globalSecondaryIndexName = "PK_FACTURA_INDEX", attributeName = "ClaveFactura")
 private String claveFactura;
```
The problem is in this class : `com.amazonaws.services.dynamodbv2.datamodeling. StandardModelFactories` and in the method `**rulesof**` and in this line because `TypeConverterFactory` from `DynamoDBMapperConfig` is `null`

```java
final DynamoDBTypeConverterFactory.Builder scalars = config.getTypeConverterFactory().override();
```

So I create the TypeConvertFactory when I am creating the amazon client inside config objet but when i am debuging in the constructor of `org.socialsignin.spring.data.dynamodb.core.DynamoDBTemplate` `TypeConverterFactory` from `dynamoDBMapperConfig` is `null` 

Regards.

**Comment by fpoolev** [comment]
This looks like the same issue as #57. I'm also trying to figure out the root cause of why the TypeConverterFactory inside the DynamoDBMapperConfig is null.

**Comment by pluttrell** [comment]
@fpoolev Any updates on this issue?

I think this is a different issue than #57 because I can work around #57 by specifying the ConversionSchema as follows:

```
DynamoDBMapperConfig.builder()
        .withConversionSchema(ConversionSchemas.V2)
        .build();
```

**Comment by cthiebault** [comment]
I also have this problem with a repository find method on global index...

**Comment by derjust** [comment]
Same root cause as #57 / #147 : Setting a half-initialized `DynamoMapperDBConfig` causing `NullPointerException` later on

Will be fixed in `5.0.3`
Already available via `5.0.3-SNAPSHOT`

**Comment by sfali** [comment]
I am also having this issue, any plan when to release `5.0.3`.
